fix: clarify documentation process (#1410)
redhat-developer/app-services-cli: CONTRIBUTING.md

## Internationalization

All text strings are placed in `./pkg/core/localize/locales/en` directory containing `.toml` files.

> This files are used in:

Reviewer: ```suggestion
These files are used in:
```

 - CLI itself - all printed messages/strings 
 - generation of the downstream and upstream documentation

This directory contains number of `toml` files that are used for:

1. Command structure that is later generated as documentation
2. Command output and error messages returned by specific commands that aren't included in the generated documentation.

> Each time we change any strings in comman structure (1) we should regenerate markdown documentation files and push them with the PR.

Reviewer: ```suggestion
Each time we change any strings in command structure (1) we should regenerate markdown documentation files and push them with the PR.
```

Downstream changes are regenerated on release. 

## Documentation

> The main CLI documentation source files are stored in the `./pkg/corelocalize/locales/en/cmd/` directory.

Author: ```suggestion
The main CLI documentation source files are stored in the `./pkg/core/localize/locales/en/cmd/` directory.
```
